workitem_blackbox_test.go

```diff
@@ -606,3 +642,433 @@ func TestPagingDefaultAndMaxSize(t *testing.T) {
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller)
+	payload := app.CreateWorkItemPayload{
+		Type: models.SystemBug,
+		Fields: map[string]interface{}{
+			models.SystemTitle: "Test WI",
+			models.SystemState: "closed"},
+	}
+
+	_, wi := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload)
+
+	defer test.DeleteWorkitemOK(t, svc.Context, svc, controller, wi.ID)
+
+	minimumPayload := getMinimumRequiredUpdatePayload(wi)
+	modifiedDescription := "Only Description is modified"
+	minimumPayload.Data.Attributes[models.SystemDescription] = modifiedDescription
+
+	controller2 := NewWorkitem2Controller(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller2)
+
+	_, updatedWI := test.UpdateWorkitem2OK(t, svc.Context, svc, controller2, wi.ID, minimumPayload)
+	require.NotNil(t, updatedWI)
+	assert.Equal(t, updatedWI.Data.Attributes[models.SystemDescription], modifiedDescription)
+}
+
+// Setup only Assignee
+func TestUpdateOnlyAssignee(t *testing.T) {
+	resource.Require(t, resource.Database)
+	pub, _ := almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("TestUpdateWI2-Service", almtoken.NewManager(pub, priv), account.TestIdentity)
+	assert.NotNil(t, svc)
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller)
+	payload := app.CreateWorkItemPayload{
+		Type: models.SystemBug,
+		Fields: map[string]interface{}{
+			models.SystemTitle: "Test WI",
+			models.SystemState: "closed"},
+	}
+
+	_, wi := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload)
+
+	defer test.DeleteWorkitemOK(t, svc.Context, svc, controller, wi.ID)
+
+	minimumPayload := getMinimumRequiredUpdatePayload(wi)
+	minimumPayload.Data.Relationships = &app.WorkItemRelationships{}
+
+	tempUser := createOneRandomUserIdentity(svc.Context, DB)
+	assert.NotNil(t, tempUser)
+
+	tempUserUUID := tempUser.ID.String()
+	assignee := &app.RelationAssignee{
+		Data: &app.AssigneeData{
+			ID:   &tempUserUUID,
+			Type: "identities",
+		},
+	}
+	minimumPayload.Data.Relationships.Assignee = assignee
+
+	controller2 := NewWorkitem2Controller(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller2)
+
+	_, updatedWI := test.UpdateWorkitem2OK(t, svc.Context, svc, controller2, wi.ID, minimumPayload)
+	require.NotNil(t, updatedWI)
+	assert.Equal(t, *updatedWI.Data.Relationships.Assignee.Data.ID, tempUserUUID)
+
+	// retrieve work item from Db and verify
+	db := gormapplication.NewGormDB(DB)
+	wiFromDB, _ := db.WorkItems().Load(svc.Context, updatedWI.Data.ID)
+	require.NotNil(t, wiFromDB)
+	assert.Equal(t, wiFromDB.Fields[models.SystemAssignee], tempUserUUID)
+}
+
+// Removing assignee
+func TestUpdateRemoveAssignee(t *testing.T) {
+	resource.Require(t, resource.Database)
+	pub, _ := almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
```

Comment:
Please don't repeat so much setup and use a test suite instead. See `workitemtype_blackbox_test.go` for an example.